modules/space/translation/horizonstranslation.cpp

```diff
@@ -186,4 +219,77 @@ void HorizonsTranslation::readHorizonsTextFile(const std::string& horizonsTextFi
     fileStream.close();
 }
 
+bool HorizonsTranslation::loadCachedFile(const std::string& file) {
+    std::ifstream fileStream(file, std::ifstream::binary);
+
+    if (!fileStream.good()) {
+        LERROR(fmt::format("Error opening file '{}' for loading cache file", file));
+        return false;
+    }
+
+    // Read how many values to read
+    int32_t nKeyframes = 0;
+
+    fileStream.read(reinterpret_cast<char*>(&nKeyframes), sizeof(int32_t));
+    if (nKeyframes == 0) {
+        throw ghoul::RuntimeError("Error reading cache: No values were loaded");
+    }
+
+    // Read the values in same order as they were written
+    for (int i = 0; i < nKeyframes; ++i) {
+        double timestamp, x, y, z;
+        glm::dvec3 gPos;
+
+        // Timestamp
+        fileStream.read(reinterpret_cast<char*>(&timestamp), sizeof(double));
+
+        // Position vector components
+        fileStream.read(reinterpret_cast<char*>(&x), sizeof(double));
+        fileStream.read(reinterpret_cast<char*>(&y), sizeof(double));
+        fileStream.read(reinterpret_cast<char*>(&z), sizeof(double));
+
+        // Recreate the position vector
+        gPos = glm::dvec3(x, y, z);
+
+        // Add keyframe in timeline
+        _timeline.addKeyframe(timestamp, std::move(gPos));
+    }
+
+    return fileStream.good();
+}
+
+void HorizonsTranslation::saveCachedFile(const std::string& file) const {
+    std::ofstream fileStream(file, std::ofstream::binary);
+    if (!fileStream.good()) {
+        LERROR(fmt::format("Error opening file '{}' for save cache file", file));
+        return;
+    }
+
+    // Write how many values are to be written
+    int32_t nKeyframes = static_cast<int32_t>(_timeline.nKeyframes());
+    if (nKeyframes == 0) {
+        throw ghoul::RuntimeError("Error writing cache: No values were loaded");
+    }
+    fileStream.write(reinterpret_cast<const char*>(&nKeyframes), sizeof(int32_t));
+
+    // Write data
+    std::deque<Keyframe<glm::dvec3>> keyframes = _timeline.keyframes();
+    for (int i = 0; i < nKeyframes; i++) {
+        // First write timestamp
+        fileStream.write(reinterpret_cast<const char*>(&keyframes[i].timestamp),
```

Comment:
In this case (and the ones below),
the `&keyframes[i]...` part would be on a new line (like the `sizeof`)